pkg/master/master.go

```diff
@@ -78,13 +80,15 @@ const (
 
 // Config is a structure used to configure a Master.
 type Config struct {
-	EtcdHelper        tools.EtcdHelper
-	EventTTL          time.Duration
-	MinionRegexp      string
-	KubeletClient     client.KubeletClient
-	PortalNet         *net.IPNet
-	EnableLogsSupport bool
-	EnableUISupport   bool
+	EtcdHelper    tools.EtcdHelper
+	EventTTL      time.Duration
+	MinionRegexp  string
+	KubeletClient client.KubeletClient
+	PortalNet     *net.IPNet
+	// allow downstream consumers to disable the core controller loops
+	EnableCoreControllers bool
```

Comment:
Didn't want to duplicate the core controller run.start in all of the possible integration tests.  The use case is Openshift, which wants to get the core controllers object and then tweak a couple of things before calling Start.  No other use case beyond that.

> On May 7, 2015, at 2:55 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/master/master.go:
> 
> > @@ -78,13 +80,15 @@ const (
> >  
> >  // Config is a structure used to configure a Master.
> >  type Config struct {
> > -	EtcdHelper        tools.EtcdHelper
> > -	EventTTL          time.Duration
> > -	MinionRegexp      string
> > -	KubeletClient     client.KubeletClient
> > -	PortalNet         *net.IPNet
> > -	EnableLogsSupport bool
> > -	EnableUISupport   bool
> > +	EtcdHelper    tools.EtcdHelper
> > +	EventTTL      time.Duration
> > +	MinionRegexp  string
> > +	KubeletClient client.KubeletClient
> > +	PortalNet     *net.IPNet
> > +	// allow downstream consumers to disable the core controller loops
> > +	EnableCoreControllers bool
> When is this ever not true? Isn't the more common pattern to have the caller call .Run() or something?
> 
> —
> Reply to this email directly or view it on GitHub.
>